module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordTest.php

```diff
@@ -502,6 +502,35 @@ public function testGetLinkDetailsFailure()
         );
     }
 
+    /**
+     * Test getLinkDetails with duplicate URLs
+     *
+     * @return void
+     */
+    public function testGetLinkDetailsWithDuplicateURLs()
+    {
+        $driver = new \VuFindTest\RecordDriver\TestHarness();
+        $driver->setRawData(
+            [
+                'URLs' => [
+                    ['desc' => 'link 1', 'url' => 'http://foo/baz1'],
+                    ['desc' => 'link 2', 'url' => 'http://foo/baz2'],
+                    ['desc' => 'link 1', 'url' => 'http://foo/baz1'],
+                    ['desc' => 'link 3', 'url' => 'http://foo/baz1']
```

Comment:
Is it intentional that the 'url' value here is a duplicate? I think that's a useful thing to test, but I wonder if that's the case, it would be worth having a 'desc' like 'link 1 (alternate description)' to make the intention of the test case more clear. It might also be interesting to add a `['url' => 'http://foo/baz1']` entry in the array to exercise the code that injects a description when it is missing, just to further demonstrate that this works correctly in combination with pre-processing.